Code/GraphMol/FileParsers/CDXMLParser.cpp

```diff
@@ -0,0 +1,666 @@
+#include "MolFileStereochem.h"
+#include <RDGeneral/FileParseException.h>
+
+// TODO
+//  I'm currently using atom map numbers to join structures
+//   this is not very forward thinking as chemdraw can also have
+//   atom map numbers in addition to the ids in the xml file.
+//  [] enhance molzip to be able to zip on on an atom tag
+//  [] what happens to stereo chemistry when the fuse_frags below is called
+//    get cases in the "wild" and see what the expectation is.
+//  [] add coordinates for the atoms.  Might not work for nicknames like Boc?
+
+using boost::property_tree::ptree;
+namespace RDKit {
+
+namespace {
+const std::string NEEDS_FUSE("CDXML_NEEDS_FUSE");
+const std::string CDXML_FRAG_ID("CDXML_FRAG_ID");
+const std::string FUSE_LABEL("CDXML_NODE_ID");
+const std::string CDX_SCHEME_ID("CDX_SCHEME_ID");
+const std::string CDX_STEP_ID("CDX_STEP_ID");
+const std::string CDX_REAGENT_ID("CDX_REAGENT_ID");
+const std::string CDX_PRODUCT_ID("CDX_PRODUCT_ID");
+const std::string CDX_AGENT_ID("CDX_AGENT_ID");
+const std::string CDX_ATOM_POS("CDX_ATOM_POS");
+const std::string CDX_ATOM_ID("_CDX_ATOM_ID");
+const std::string CDX_BOND_ID("_CDX_BOND_ID");
+const std::string CDX_BOND_ORDERING("CDXML_BOND_ORDERING");
+
+struct BondInfo {
+    int bond_id;
+    int start;
+    int end;
+    Bond::BondType order;
+    std::string display;
+    Bond::BondType getBondType() {
+        return order;
+    }
+    bool validate(const std::map<unsigned int, Atom *> &ids, unsigned int num_atoms) const {
+        auto s = ids.find(start);
+        auto e = ids.find(end);
+        if (s==ids.end() || e==ids.end()) {
+            return false;
+        }
+        auto st = s->second->getIdx();
+        auto et = e->second->getIdx();
+        
+       return (st >=0 && st < num_atoms &&
+                     et >=0 && et < num_atoms &&
+                     st != et);
+    }
+};
+
+struct StereoGroupInfo {
+    int sgroup = -1;
+    bool conflictingSgroupTypes = false;
+    RDKit::StereoGroupType grouptype;
+    std::vector<Atom*> atoms;
+};
+
+struct SchemeInfo {
+    int scheme_id;
+    int step_id;
+    std::vector<unsigned int> ReactionStepProducts;
+    std::vector<unsigned int> ReactionStepReactants;
+    std::vector<unsigned int> ReactionStepObjectsAboveArrow;
+    std::vector<unsigned int> ReactionStepObjectsBelowArrow;
+    std::vector<unsigned int> ReactionStepAtomMap;
+};
+
+unsigned int get_fuse_label(Atom*atm) {
+    // return atm->getAtomMapNum(); easier debugging
+    unsigned int label = 0; // default is no label
+    atm->getPropIfPresent<unsigned int>(FUSE_LABEL, label);
+    return label;
+}
+
+void set_fuse_label(Atom*atm, unsigned int idx) {
+    //atm->setAtomMapNum(idx); //for debugging
+    if(idx) {
+        atm->setProp<unsigned int>(FUSE_LABEL, idx);
+    } else {
+        atm->clearProp(FUSE_LABEL);
+    }
+}
+
+template <typename Q>
+Atom * addquery(Q *qry, std::string symbol, RDKit::RWMol &mol, unsigned int idx) {
```

Comment:
you have multiple gratuitous `RDKit::` specfications in here that I don't think are necessary